src/config_parser.ml

```diff
@@ -120,6 +138,62 @@ let pp_dhcp_range ppf
     Fmt.(option ~none:nop (any "," ++ pp_duration))
     lease_time
 
+let pp_dhcp_config ppf
+    { id; nets; tags; macs; ipv4; ipv6; lease_time; ignore; domain_name } =
+  let sep =
+    let use_sep = ref false in
+    fun () ->
+      if !use_sep then Fmt.pf ppf ",";
+      use_sep := true
+  in
+  List.iter
+    (fun mac ->
+      sep ();
+      Macaddr.pp ppf mac)
+    macs;
+  Option.iter
+    (fun id ->
+      sep ();
+      match id with
+      | `Any_client_id -> Fmt.pf ppf "id:*"
+      | `Client_id id ->
+          (* TODO: use hex when text is inappropriate *)
+          Fmt.pf ppf "id:%s" id)
+    id;
+  List.iter
+    (fun net ->
+      sep ();
+      Fmt.pf ppf "net:%s" net)
+    nets;
```

Comment:
as noted below, I think we need `set:<tag>` -- and don't quite understand where `net:` comes from (unable to find in example or manpage).

Reply:
This paragraph from the NOTES section documents it:

       This  system evolved from an earlier, more limited one and for backward
       compatibility "net:" may be used instead of "tag:" and  "set:"  may  be
       omitted.  (Except  in  --dhcp-host, where "net:" may be used instead of
       "set:".) For the same reason, '#' may be used instead of '!'  to  indi‐
       cate NOT

Reply:
I'd omit the `net:`, as well as `#` -- and write that in the README. Let's use the latest configuration format version, and if someone hits such an error, we can help them to move their configuration to the latest specification.

Reply:
At this point I don't do any processing of tags, i.e. no special handling of `!` nor `#`. I think I will defer that for another time when I have a better picture on the tag mechanism in my head.